test/integration/document_test.ts

```diff
@@ -931,4 +932,17 @@ describe('Document', function () {
       assert.equal(doc.toSortedJSON(), '{"counter":100}');
     });
   });
+
+  it('deregisterElement must not deregister same element twice in nested object', async function ({ task }) {
+    type TestDoc = { shape?: { point?: { x?: number; y?: number } } };
+    const docKey = toDocKey(`${task.name}-${new Date().getTime()}`);
+    const doc = new yorkie.Document<TestDoc>(docKey);
+
+    doc.update((root) => {
+      root.shape = { point: { x: 0, y: 0 } };
+      delete root.shape;
+    });
+    assert.equal(doc.getGarbageLen(), 4); // shape, point, x, y
+    assert.equal(doc.garbageCollect(MaxTimeTicket), 4); // The number of GC nodes must also be 4.
+  });
```

Comment:
It would be good to move this to a `gc_test.ts`, and rename test to `garbage collection test for nested object`.

Reply:
OK. I'll move it to `gc_test.ts`.